src/main/java/me/minidigger/hangar/controller/PagesController.java

```diff
@@ -1,45 +1,155 @@
+import me.minidigger.hangar.model.viewhelpers.ScopedProjectData;
+import me.minidigger.hangar.service.MarkdownService;
+import me.minidigger.hangar.service.project.PagesFactory;
+import me.minidigger.hangar.service.project.PagesSerivce;
+import me.minidigger.hangar.service.project.ProjectService;
+import me.minidigger.hangar.util.RouteHelper;
+import me.minidigger.hangar.util.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.configurationprocessor.json.JSONException;
+import org.springframework.boot.configurationprocessor.json.JSONObject;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.lang.Nullable;
 import org.springframework.security.access.annotation.Secured;
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.server.ResponseStatusException;
 import org.springframework.web.servlet.ModelAndView;
-
-import me.minidigger.hangar.controller.HangarController;
+import org.springframework.web.servlet.view.RedirectView;
 
 @Controller
 public class PagesController extends HangarController {
 
-    @PostMapping("/pages/preview")
-    public Object showPreview() {
-        return null; // TODO implement showPreview request controller
+    private final HangarConfig hangarConfig;
+    private final ProjectService projectService;
+    private final PagesSerivce pagesSerivce;
+    private final PagesFactory pagesFactory;
+    private final HangarDao<ProjectPageDao> projectPageDao;
+    private final RouteHelper routeHelper;
+    private final MarkdownService markdownService;
+
+    @Autowired
+    public PagesController(HangarConfig hangarConfig, ProjectService projectService, PagesSerivce pagesSerivce, PagesFactory pagesFactory, HangarDao<ProjectPageDao> projectPageDao, RouteHelper routeHelper, MarkdownService markdownService) {
+        this.hangarConfig = hangarConfig;
+        this.projectService = projectService;
+        this.pagesSerivce = pagesSerivce;
+        this.pagesFactory = pagesFactory;
+        this.projectPageDao = projectPageDao;
+        this.routeHelper = routeHelper;
+        this.markdownService = markdownService;
+    }
+
+    @PostMapping(value = "/pages/preview", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public Object showPreview(@RequestBody String raw) throws JSONException {
+        JSONObject rawJson;
+        try {
+            rawJson = new JSONObject(raw);
+            rawJson.getString("raw");
```

Comment:
Is this here to catch an exception? Or why is the result of getstring ignored here?

Reply:
Yeah, getString throws a exception that will be cause if it doesn't exist.